src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -1747,10 +1759,15 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
                 ptr_loads = ptr_loads_copy;
                 indices.push_back(tmp);
             }
-            if (ASRUtils::expr_type(x.m_v)->type == ASR::ttypeType::Pointer) {
+            int a_kind = ASRUtils::extract_kind_from_ttype_t(x_mv_type);
+            bool is_i8_array = (ASR::is_a<ASR::Integer_t>(*x_mv_type) && a_kind == 1 &&
+                                ASRUtils::expr_abi(x.m_v) == ASR::abiType::BindC);
+            if (ASR::is_a<ASR::Pointer_t>(*x_mv_type) ||
+               (is_i8_array && ASR::is_a<ASR::StructInstanceMember_t>(*x.m_v))) {
                 array = CreateLoad(array);
             }
             bool is_data_only = is_argument && !ASRUtils::is_dimension_empty(m_dims, n_dims);
+            is_data_only = is_data_only || is_i8_array;
```

Comment:
In here I am not sure we should special case i8. I think there is nothing special about i8, we should have exactly the same handling for all integer (and real and complex) arrays.

Reply:
Well, I guess the requirement was to create data only array for i8 if its a BindC. See https://github.com/lcompilers/lpython/issues/1333